mbed_greentea/tests_spec.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2015 ARM Limited, All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains classes to represent Test Specification interface that defines the data to be generated by/from
+a build system to give enough information to Greentea.
+"""
+
+
+class TestBinary:
+    """
+    Class representing a Test Binary.
+    """
+    def __init__(self, path, flash_method):
+        """
+        ctor.
+
+        :return:
+        """
+        self.path = path
+        # attributes
+        self.flash_method = flash_method
+
+    def get_path(self):
+        """
+        Gives binary path.
+        :return:
+        """
+        return self.path
+
+
+class Test:
+    """
+    class representing a Test artifact that may contain more than one test binaries.
+    """
+    def __init__(self, name):
+        """
+        ctor.
+
+        :param name:
+        :return:
+        """
+        self.name = name
+        self.binaries = {}
+
+    def get_name(self):
+        """
+        Gives test name.
+
+        :return:
+        """
+        return self.name
+
+    def get_binaries(self):
+        """
+        Gives test binaries.
+        :return:
+        """
+        return self.binaries
+
+    def get_binary(self, type):
+        """
+        Gives a test binary of specific flash type.
+
+        :param type:
+        :return:
+        """
+        return self.binaries[type]
+
+    def parse(self, test_json):
+        """
+        Parse json contents into object.
+
+        :param test_json:
+        :return:
+        """
+        print test_json
+        for _, binary in test_json['binaries'].iteritems():
+            tb = TestBinary(binary['path'], binary['flash_method'])
+            self.binaries[binary['flash_method']] = tb
+
+    def add_binary(self, name, path, flash_method):
+        """
+        Add binary to the test.
+
+        :param name:
+        :param path:
+        :param flash_method:
+        :return:
+        """
+        self.binaries[name] = TestBinary(path, flash_method)
+
+
+class TestBuild:
+    """
+    class for Test build.
+    """
+
+    def __init__(self, name, platform, toolchain, baud_rate, base_path):
+        """
+        ctor.
+
+        :param platform:
+        :param toolchain:
+        :return:
+        """
+        self.name = name
```

Comment:
You have lots of these getters for these fields. You wanted to make them private and accesible via get_ only?
If so let's make it like this:
```python
        self.__name = name
        self.__platform = platform
        self.__toolchain = toolchain
        self.__baud_rate = baud_rate
        self.__base_path = base_path
```